src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -264,10 +311,12 @@ public Response downloadZip(JsonObject content,
             Result result = controller.execute();
             if (result instanceof Failed)
             {
-               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(result.getMessage()).build();
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(getMessage(result)).build();
             }
             else
             {
+               if (exposedCommands.generateZipCommand(commandName))
```

Comment:
So 2 different methods on the resource; one for zip-creating commands and one for non-zip creating commands right? 

Should they then have different URIs? 

The only downside is then that the client is gonna have to change to be aware of which command names are zip-creating versus which are not. Though I guess the front end needs that information anyways as it needs to know if it should open the result as a zip or just go to the usual finished page?